app/src/main/java/com/example/teamproject/MyLocation.java

```
@@ -0,0 +1,124 @@
+package com.example.teamproject;
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class MyLocation extends Location{
+    Timer timer1;
+    LocationManager lm;
+    LocationResult locationResult;
+    boolean gps_enabled=false;
+    boolean network_enabled=false;
+
+//    public MyLocation {
+//        //empty constructor
+//    }
+
+
+    public MyLocation(String provider, Timer timer1, LocationManager lm, LocationResult locationResult, boolean gps_enabled, boolean network_enabled, LocationListener locationListenerGps, LocationListener locationListenerNetwork) {
```

There is a lot to digest here.

Will `gps_enabled` or `network_enabled` ever be passed in? If not, I would keep them private and initialized in the constructor. 

Constructor -> checkNetworkProviderPermissions(); -> checkGPSProviderPermissions(); -> 
try { <check provider> } catch (<provider not available exception> $e) .... { }

It is bad practice to swallow (read: blanket catch) all exceptions. Doing so would hide any failures you are not expecting to catch and can be excruciating to debug.